pytket/extensions/azure/backends/azure.py

```diff
@@ -185,6 +195,16 @@ def __init__(
         self._result_bits: dict[ResultHandle, list] = {}
         self._result_c_regs: dict[ResultHandle, list] = {}
 
+        self._device_type = DeviceType.Default
+
+        if self._backendinfo.device_name:
+            if self._backendinfo.device_name[:11] == "quantinuum.":
+                self._device_type = DeviceType.Quantinuum
+            elif self._backendinfo.device_name[:5] == "ionq.":
+                self._device_type = DeviceType.Ionq
+            elif self._backendinfo.device_name[:8] == "rigetti.":
+                self._device_type = DeviceType.Rigetti
+
```

Comment:
I think it would be good to emit a warning if none of these conditions is satisfied.

Reply:
I think we should also emit the warning if `f self._backendinfo.device_name` is `None`, i.e. whenever we settle on `DeviceType.Default`. In case there is a `device_name` but it doesn't match our patterns, we should include the name in the warning message.

Reply:
Done in [7aaf7f5](https://github.com/CQCL/pytket-azure/pull/32/commits/7aaf7f5414da3ca9dfa1eef592592d4fb23b44b7)